node/clients/masterdata.ts

```diff
@@ -0,0 +1,129 @@
+  if (status === 401) {
+    throw new AuthenticationError(e)
+  }
+  if (status === 403) {
+    throw new ForbiddenError(e)
+  }
+  if (status === 400) {
+    throw new UserInputError(e)
+  }
+  throw e
+}
+
+const paginationArgsToHeaders = ({page, pageSize}: PaginationArgs) => {
+  if (page < 1) {
+    throw new UserInputError('Smallest page value is 1')
+  }
+  const startIndex = (page - 1) * pageSize
+  const endIndex = startIndex + pageSize
+  return {
+    'REST-Range': `resources=${startIndex}-${endIndex}`
+  }
+}
+
+export class MasterData extends JanusClient {
+  constructor (ctx: IOContext, options?: InstanceOptions) {
+    super(ctx, {
+      ...options,
+      headers: {
+        ...options && options.headers,
+        ...{Accept: 'application/vnd.vtex.ds.v10+json'},
+        ...ctx.adminUserAuthToken ? {VtexIdclientAutCookie: ctx.adminUserAuthToken} : null,
+        ...ctx.storeUserAuthToken ? {[`VtexIdclientAutCookie_${ctx.account}`]: ctx.storeUserAuthToken} : null,
+      },
+    })
+  }
+
+  public getDocument = (acronym: string, id: string, fields: string[]) => this.get<Document>(
+    this.routes.document(acronym, id),
+    { metric: 'masterdata-getDocument', params: { _fields: fields } }
+  )
+
+  public createDocument = (acronym: string, fields: KeyValue[]) => this.post<Document>(
+    this.routes.documents(acronym),
+    { metric: 'masterdata-createDocument', params: parseFieldsToJson(fields) }
+  )
+
+  public updateDocument = (acronym: string, id: string, fields: KeyValue[]) => this.patch<Document>(
+    this.routes.document(acronym, id),
+    { metric: 'masterdata-updateDocument', params: parseFieldsToJson(fields) }
+  )
+
+  public searchDocuments = (acronym: string, fields: string[], where: string, pagination: PaginationArgs) => this.get<any[]>(
+    this.routes.search(acronym),
+    { headers: paginationArgsToHeaders(pagination) , metric: 'masterdata-searchDocuments', params: { _fields: fields, _where: where } }
+  )
+
+  public deleteDocument = (acronym: string, id: string) => this.delete(
+    this.routes.document(acronym, id),
+    {metric: 'masterdata-deleteDocument'}
+  )
+
+  public uploadAttachment = (acronym: string, id: string, fields: string, formData: FormData) => this.post<any>(
+    this.routes.attachments(acronym, id, fields),
+    formData,
+    { headers: formData.getHeaders(), metric: 'masterdata-uploadAttachment'}
+  )
+
+    protected get = <T>(url: string, config?: RequestConfig) => {
+      return this.http.get<T>(url, config).catch(statusToError)
+    }
+
+    protected post = <T>(url: string, data?: any, config?: RequestConfig) => {
+      return this.http.post<T>(url, data, config).catch(statusToError)
+    }
+
+    protected delete = <T>(url: string, config?: RequestConfig) => {
+      return this.http.delete<T>(url, config).catch(statusToError)
+    }
+
+    protected patch = <T>(url: string, config?: RequestConfig) => {
+      return this.http.patch<T>(url, config).catch(statusToError)
```

Comment:
error here: the definition of patch is:
`patch: <T = void>(url: string, data?: any, config?: RequestConfig) => Promise<T>;`

so patch should be just like this.post, you are not passing data.

Reply:
fixed